src/librustc_trans/back/link.rs

```diff
@@ -692,6 +692,10 @@ fn link_natively(sess: &Session,
             info!("linker stdout:\n{}", escape_string(&prog.stdout[..]));
         },
         Err(e) => {
+            // Trying to diagnose https://github.com/rust-lang/rust/issues/33844
+            if cfg!(windows) && e.raw_os_error() == Some(206) {
+                sess.note_without_error(&format!("linker command: {:?}", cmd));
+            }
```

Comment:
Above it looks like there's a `sess.struct_err(..).note(..)`, perhaps this could follow the same format as that? It may also be worth doing this on all platforms for all errors, it's arguably information that shouldn't be hidden